stdlib/src/utils/static_tuple.mojo

```diff
@@ -254,3 +314,139 @@ struct StaticTuple[element_type: AnyRegType, size: Int](Sized):
         var base_ptr = Pointer[Self.type].address_of(self.array).address
         var ptr = __mlir_op.`pop.array.gep`(base_ptr, Int(0).value)
         return Pointer(ptr)
+
+
+# ===----------------------------------------------------------------------===#
+# Array
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct Array[ElementType: CollectionElement, size: Int](Sized):
+    """A statically sized tuple type which contains elements of homogeneous types.
+
+    Parameters:
+        ElementType: The type of the elements in the tuple.
+        size: The size of the tuple.
+    """
+
+    alias type = __mlir_type[
+        `!pop.array<`, size.value, `, `, Self.ElementType, `>`
+    ]
+    var array: Self.type
+    """The underlying storage for the static tuple."""
```

Comment:
**Suggestion** `s/static tuple/array`.  Ditto elsewhere in this struct definition - essentially let's use the word "array" instead of "tuple" - WDYT?